### PR TITLE
fix(sbb-train-formation): fix accessibility and overlapping sectors

### LINKED ISSUE #1570: [Bug]: sbb-train-formation - display two trains with overlapping sectors is not possible
currently the sbb-sector wrappes sbb-wagons. Which limits to render some cases with "mounted trains".

It should be possible to display one sector with trains with multiple direction:

Example:
- SectorA contains: train1, train2 and train3
- SectorB contains: train3, trainX

-> train1 - train2 have diffrent direction and train3 - trainX have an other direction.

So component should also work without sbb-sector and there should be an other solution for displaying sectors (Each sbb-wagon and the block-component should contain the info for the sector)

### LINKED ISSUE #1638: [Bug]: sbb-train-formation - semantic HTML and a11y
We got few feedback for the sbb-train-formation:

- If there no attribute icons or occupancy icons, there is still some markup or some text for screenreader
- the whole component should be a list where a sbb-wagon should be a list item
- the label for sector and direction are currently a h3. (should be lower or customizable)

image

### PR DESCRIPTION
**Pull request by jeripeierSBB**
Closes #1570 
Closes #1638

BREAKING CHANGE:
- Component `sbb-wagon` was renamed to `sbb-train-wagon`
- Component `sbb-wagon-blocked-passage` was renamed to `sbb-train-blocked-passage`
- Component `sbb-sector` was completely removed. Now you have to set the sector as property / attribute on `sbb-train-wagon` and additionally now only the letter should be provided and not the word "sector".
- Property `customAccessibilityText` of `sbb-train-wagon` has been removed without replacement.

### CODE REVIEW

**Comment by federicoisepponfincons** [comment]
Should the train sectors expand their width in order to adapt to the train length?

Like adding `flex-grow: 1;` to `.sbb-train-formation__sector`, but it also needs to limit the max width

**Review by kyubisation — CHANGES_REQUESTED** [review]
Nice work! 😃 
Some small details need to be solved

File: src/components/sbb-train-formation/sbb-train-formation.tsx [hunk 1]
```diff
@@ -21,20 +40,24 @@ export class SbbTrainFormation {
...
-  public disconnectedCallback(): void {
-    this._contentResizeObserver.disconnect();
+  public connectedCallback(): void {
+    this._handleSlotChange();
   }
 
-  private _onResize(): void {
+  public componentDidLoad(): void {
     this._contentWidth = this._formationDiv.getBoundingClientRect().width;
+    this._contentResizeObserver.observe(this._formationDiv);
```
[the remark this answers is not in the document]
Comment (jeripeierSBB) [reply src/components/sbb-train-formation/sbb-train-formation.tsx:51]: was an old leftover, corrected :)

File: src/components/sbb-train-formation/sbb-train-formation.tsx [hunk 2]
```diff
@@ -46,15 +69,117 @@ export class SbbTrainFormation {
...
+        }
+
+        return collected;
+      },
+      []
+    );
+  }
+
+  private _readTrains(): void {
+    this._trains = Array.from(this._element.children).filter(
```
[the remark this answers is not in the document]
Comment (jeripeierSBB) [reply src/components/sbb-train-formation/sbb-train-formation.tsx:123]: I prefer :scope > ..., changed all occurrences

File: src/components/sbb-navigation-marker/sbb-navigation-marker.tsx [hunk 3]
```diff
@@ -90,9 +90,9 @@ export class SbbNavigationMarker implements ComponentInterface {
 
   // Create an array with only the sbb-navigation-action children.
   private _readActions(): void {
-    this._actions = Array.from(this._element.children).filter(
-      (e): e is HTMLSbbNavigationActionElement => e.tagName === 'SBB-NAVIGATION-ACTION'
-    );
+    this._actions = Array.from(
+      this._element.querySelectorAll('sbb-navigation-action')
```
Comment (kyubisation) [inline src/components/sbb-navigation-marker/sbb-navigation-marker.tsx:94]: This is working?
Just making sure. 😄

File: src/components/sbb-train-formation/sbb-train-formation.tsx [hunk 4]
```diff
@@ -11,50 +19,143 @@ import { AgnosticResizeObserver as ResizeObserver } from '../../global/helpers/r
...
+    this._formationDiv = el;
+    this._contentResizeObserver.observe(this._formationDiv);
   }
 
   public render(): JSX.Element {
+    // We should avoid lists with only one entry
+    if (this._trains?.length > 1) {
+      this._trains.forEach((train, index) => train.setAttribute('slot', `train-${index}`));
+    } else {
+      this._trains.forEach((train) => train.removeAttribute('slot'));
```
Comment (kyubisation) [inline src/components/sbb-train-formation/sbb-train-formation.tsx:119]:
```suggestion
      this._trains?.forEach((train) => train.removeAttribute('slot'));
```

**Review by kyubisation — APPROVED** [review]
LGTM 👍

File: src/components/sbb-navigation-marker/sbb-navigation-marker.tsx (hunk 3 above)
Comment (jeripeierSBB) [reply src/components/sbb-navigation-marker/sbb-navigation-marker.tsx:94]: I thought I reverted this change. thanks for catching :)

**Pull request merged**